R/regression.R

```diff
@@ -358,3 +358,76 @@ rae <- function(actual, predicted) {
 explained_variation <- function(actual, predicted) {
     1 - rse(actual, predicted)
 }
+
+#' Mallows's Cp
+#' 
+#' \code{mallowsCp} computes the Mallows's Cp statistic for a linear regression
+#' model. Mallows's Cp is used to assess the fit of an OLS model, often in order
+#' to find the best subsets of the model's predictors. A smaller value of Cp
+#' is preferred, although Cp ~= p is generally accepted. Mallows's Cp is computed
+#' as follows:
+#' 
+#' \deqn{\frac{SSE_\text{sub}}{MSE_\text{full}} - n + 2p_\text{sub}}
+#' 
+#' or, an alternative definition is occasionally given as:
+#' 
+#' \deqn{MSE_\text{sub} + 2p_\text{sub}MSE_\text{full}}
+#' 
+#' While the two definitions don't give the same answer, a model with the lowest
+#' Cp will have the lowest Cp using both definitions.
+#'
+#' @param full_model Either a \code{formula} or \code{lm} for the OLS model using all predictors.
+#' @param sub_model Either a \code{formula} or \code{lm} for the OLS model with a subset of predictors
+#' @param data The data (only necessary whe supplying formulae)
+#' @param alt_definition Whether or not to use the alternate definition for Mallows's Cp
+#' @export
+mallowsCp <- function(full_model, sub_model, data = NULL, alt_definition = FALSE) {
```

Comment:
I wonder if the arguments to this function should be `actual`, `predicted_full`, `predicted_sub`, `p_full`, and `p_sub`, instead of requiring the user to pass in model objects or formulae?

The advantages of this approach are
1. It keeps consistent with the rest of the package
2. It allows the user to use whatever software they want in order to fit the regression, not necessarily `lm`,

The disadvantages of this approach are
1. An uglier function signature with more arguments.

What are your thoughts on this decision? Is there a particular reason that you designed the function the way you did?